superset/sql_lab.py

```diff
@@ -245,9 +245,6 @@ def execute_sql_statements(
     db_engine_spec = database.db_engine_spec
     db_engine_spec.patch()
 
-    if store_results and not results_backend:
-        raise SqlLabException("Results backend isn't configured.")
```

Comment:
shouldn't we still throw an error? I'd imagine stuff wouldn't work right if you're trying to store results without a backend so we shouldn't let the user even try it

Reply:
I think we could change this to `if async_ and not results_backend: raise ...`.

With this PR, now all queries have `store_results` set, since for sync queries we still want to try to store them in order to load them later. I can remove the `store_results` flag (since it's always true), and raise the exception if the query is async and there is not results backend.

Reply:
My understanding of `store_results` is that the query payload (including results) should be persisted in the `RESULTS_BACKEND`. As it stands, this is causing synchronous queries to try and persist this data, even if async queries are disabled for a database, resulting in failure.